hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java

```diff
@@ -372,6 +392,46 @@ public boolean txnCommitOffer(long itemId, Data data, boolean backup) {
         return true;
     }
 
+    public boolean txnCommitAddAll(Set<Long> itemIds, List<Data> data, boolean backup) {
+        int i = 0;
+        Map<Long, Data> toStore = new HashMap<>();
+        for (Long itemId : itemIds) {
+            Data datum = data.get(i);
+            QueueItem item = txMap.remove(itemId);
+            if (item == null && !backup) {
+                throw new TransactionException("No reserve: " + itemId);
+            } else if (item == null) {
+                item = new QueueItem(this, itemId, datum);
+            }
+            item.setData(datum);
+            if (!backup) {
+                getItemQueue().offer(item);
+                cancelEvictionIfExists();
+            } else {
+                getBackupMap().put(itemId, item);
+            }
+            i++;
+            toStore.put(itemId, datum);
+        }
+        if (store.isEnabled() && !backup) {
+            try {
+                store.storeAll(toStore);
+            } catch (Exception e) {
+                logger.warning("Exception during store", e);
+            }
+        }
+        return true;
+    }
+
+    public boolean txnCommitContainsAll(List<Data> data) {
+        for (Data datum : data) {
+            if (getItemQueue().stream().noneMatch(item -> datum.equals(item.getData()))) {
```

Comment:
why not cache `getItemQueue().stream()` ?

Reply:
Streams are not reusable

Reply:
I would think that performance is important.